### PR TITLE
Make Projects and Technologies Followable and Likable

### LINKED ISSUE #18: User like a Project

### LINKED ISSUE #19: User like a Technology

### LINKED ISSUE #20: User follow Project

### LINKED ISSUE #21: User follow Technology

### PR DESCRIPTION
**Pull request by johnvoloski**
Fixes #18 
Fixes #19
Fixes #20 
Fixes #21

* [X] Create migrations
* [X] References on user model
* [X] References on project model
* [X] References on technology model
* [X] Rspec
* [X] View/Controller

### CODE REVIEW

**Comment by dwayhs** [comment]
Changed the description of your commit to use the github checkboxes.
It's nice and shows a progress bar on the pull request index page.

:smile:

**Comment by dwayhs** [comment]
You don't need to write cucumber tests for now because the user interface is not fully specified.

We can do this later

**Comment by dwayhs** [comment]
Also added the links to the issues.
If you do this, the cards on Waffle.io are automatically moved to the 'in progress' column.

https://waffle.io/CWISoftware/enlighten

**Comment by johnvoloski** [comment]
Thanks @dwayhs. I'm out practice.

File: app/models/user.rb [hunk 1]
```diff
@@ -13,9 +13,12 @@
 #  last_sign_in_at        :datetime
 #  current_sign_in_ip     :inet
 #  last_sign_in_ip        :inet
+#  followed_technologies  :join_table Technology
```
Comment (dwayhs) [inline app/models/user.rb]: We are using the gem (annotate)[https://rubygems.org/gems/annotate].
You can just execute ```annotate``` on the command line and this comments are automatically generated for you based on the current schema.

Comment (johnvoloski) [inline app/models/user.rb]: :+1:  nice!

**Comment by johnvoloski** [comment]
@dwayhs

**Comment by dwayhs** [comment]
+1 Approve

**Comment by dwayhs** [comment]
@johnvoloski ready to merge?

**Comment by dwayhs** [comment]
I'll just rebase against the master branch and merge

**Comment by johnvoloski** [comment]
@dwayhs done!

**Pull request merged**